pkg/broker/util.go

```diff
@@ -46,20 +46,21 @@ func SpecToService(spec *apb.Spec) Service {
 		Description: spec.Description,
 		Tags:        make([]string, len(spec.Tags)),
 		Bindable:    spec.Bindable,
-		Plans:       toBrokerPlans(spec.Plans),
+		Plans:       toBrokerPlans(spec.FQName, spec.Plans),
 		Metadata:    spec.Metadata,
 	}
 
 	copy(retSvc.Tags, spec.Tags)
 	return retSvc
 }
 
-func toBrokerPlans(apbPlans []apb.Plan) []Plan {
+func toBrokerPlans(fqname string, apbPlans []apb.Plan) []Plan {
 	brokerPlans := make([]Plan, len(apbPlans))
 	i := 0
 	for _, plan := range apbPlans {
+		genID := fmt.Sprintf("%v-%v", fqname, plan.Name)
```

Comment:
Is there a character limit on these IDs and how does it compare to the char limit on the unique plan IDs making them not unique.  e.g. 
```yaml
serviceclass: dh-cfchase-reallyreallylongfqnameblahblah
  plans:
    - plan1
    - plan2
``` 
turning into 
```
dh-cfchase-reallyreallylongfqnameblahblah-plan1
dh-cfchase-reallyreallylongfqnameblahblah-plan2
```

and then getting cut short if really long to:
```
dh-cfchase-reallyreallylongfqnameblahblah-pla
dh-cfchase-reallyreallylongfqnameblahblah-pla
```

and then not being unique

Reply:
how about 
serviceclass name = 32 characters of service plan + 8 character generated (40 chars max)
plan name = 16 characters of english + 8 characters generated (24 chars max)
total of 64 characters max, pretty darn unique

Reply:
@cfchase as far as serviceclass name is concerned I think we probably want to limit the number of places that have this logic of truncating names. Currently, in the broker:

```
                re := regexp.MustCompile(fqNameRegex)
                spec.FQName = re.ReplaceAllLiteralString(
                        fmt.Sprintf("%v-%v", registryName, spec.FQName),
                        "-")
                spec.FQName = fmt.Sprintf("%.51v", spec.FQName)
                if strings.HasSuffix(spec.FQName, "-") {
                        spec.FQName = spec.FQName[:len(spec.FQName)-1]
                }
```

`registryName` is typically something like `dh` (I've used `dz` when I've added my org as an additional registry) so we would have something like `dh-etherpad-apb`. We are truncating this to 51 characters wide but, I believe, this is sufficiently unique. My only concern is, **if** we want to shorten this, we should do it here in the broker **before** we create plan names.

#### Not really related to this PR

Follow up questions regarding serviceclass names, apb names, fqnames, etc:
1. Are we doing anything to prevent someone from using the same name 2x in the broker config?
1. How do we handle multiple APB containers whose apb.yml have the same name?

Reply:
@eriknelson @djzager @dymurray @cfchase so does the ID need to be human readable? can't I just do a UUID? like f4b72d2a-bf85-4847-a19f-edd8beef0ddd

Reply:
@jmrodri I don't think there is any requirement that they are human readable. I think a UUID would work fine and in fact that is what the template broker is doing.

Reply:
+1 to @dymurray comment